Blog post: Preparing the Terrain (#2414)
mainmatter/mainmatter.com: src/posts/2024-05-21-team-buy-in.md

---

> title: "Preparing the Terrain"

Reviewer: How about: (although I am not sure the exact term is augmentation?)
```suggestion
title: "Team Buy-in: Preparing the Terrain for Augmentation"
```

Author: Yeah, I'm not sure - augmentation is part of what we do, there's assessment, strategic advice... perhaps simply consulting? I see what you mean about being clear what terrain we're actually talking about

Reviewer: We can also go in a different direction, like: "Paving the way for successful collaboration" (or engagement? What's more appropriate?). Either with "Team buy-in:" at the beginning or without.

Reviewer: Or if you like "Preparing the terrain" better we can also keep that of course

authorHandle: oliverbarnes
twitter: oliverbarnes
tags: [startups, consulting]
bio: "Oliver Azevedo Barnes"
description:
  "Oliver Barnes on the importance of getting the team onboard when bringing in a consultancy"
og:
  image: /assets/images/posts/2023-11-03-the-case-against-project-managers/og-image.jpeg
tagline: |
  <p>A consultancy engagement can be a very productive, gratifying collaborative process. That's _if_ the client team is onboard with having it come in to help</p>

image: "/assets/images/posts/2024-05-21-team-buy-in/header-visual.jpeg"
imageAlt: "A rocket icon on a grey image background"
---

Consultants get a bad rep sometimes, and often for good reason, specially due to powerful top tier consultancies involved in everything from privatization of public companies to disaster relief. They're everywhere these days.

> Boutique, technical consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.

Author: Reworded to "Software engineering consultancies"

> As @KevinBongart puts it eloquently, the perception can be like:

Reviewer: ```suggestion
As our Engineering Manager, [Kevin Bongart](https://github.com/KevinBongart), puts it eloquently, the perception can be like:
```

> 

Reviewer: @oliverbarnes this is how I'd change Kevin's quote: 

"Ugh, leadership is bringing in a bunch of consultants from Big Co to audit the team and point out our inefficiencies… this can't be good, right?"

In the team's view,  consultants are bad news: audits, inefficiencies, and changes suggested by _outsiders_! The natural reaction is to resist the implementation of all adjustments, even positive ones.
The team would need a very good reason to be on board. And that's called team buy-in.

Author: perfect - I've added it on acf90553e7d92c71f3b76f3ac058ed1765a55fe3

> As our Engineering Manager @KevinBongart puts it eloquently, the perception can be like:

Reviewer: Maybe we link to GitHub? So:
```suggestion
As our Engineering Manager [Kevin Bongart](https://github.com/KevinBongart) puts it eloquently, the perception can be like:
```

We discuss that with leadership, and different stakeholders who might be involved in bringing us, and try to work with them in preparing the terrain for the engagement.

## Happy path

When the client team is bought-in, we get an environment of trust, a sweet spot where we complement the client team seamlessly. 

Context and knowledge are shared proactively, affording us smooth onboarding. Once we have some recommendations to give, these are discussed constructively. Decisions on what to implement are naturally done in consensus.

A virtuous cycle ensues, where implementing changes is progressively faster and more effective, and ideas flow freely back and forth. Together we shift into continous improvement mode. Everybody wins.

So. Just how do you get buy-in?

## Get the team involved early on

The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea. Building consensus regarding the engagement. 

> Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to disagree and commit.

Reviewer 2: ```suggestion
Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [Disagree and Commit](https://en.wikipedia.org/wiki/Disagree_and_commit).
```

That's a very bad start for a long term relationship.

Hence, Mainmatter doesn't do audits. We offer blameless assessments as a resource to help teams plan their future work, possibly leading us to a fruitful collaboration down the line. 

### Collaboration, not intervention

In a similar fashion, when we augment a team (with engineers, architects, designers or managers, sometimes all of the above), we work hard to ensure this work is collaborative and not perceived as an intervention.

We're making suggestions, sharing knowledge and practices, pairing and planning together, constantly building consensus.

We're complementing the team until they no longer need us. 

### Alignment on scope

An engagement can take many shapes and forms. It can have a tightly bounded scope, like updating the runtime and dependencies of a given codebase, or a much broader and open ended one, as with continuous assessment of team processes and practices. Sometimes implementation work might be done in tandem with continuous assessment and counselling, too.

> Whatever the scope, it needs to be clearly communicated as soon as it is known. People can become territorial over their work. Nobody likes someone looking over their shoulders nor receiving unsolicited advice. If there's a misunderstanding over the scope of our work, that's exactly how it'll feel to the client team.

Author: I agree 100%, I don't think that's a good approach either. That's not what I'm trying to say, and I can try to word it differently if need be.

My point here specifically is about transparency of scope, specially in a low-trust environment. _How_ that's best "communicated" (not commanded), as I try to make clear overall in the blog post, is by discussing early and openly the intention and scope with the team, and then trying to get the buy-in. 

Tight scope:

"we're thinking of bringing in a team that can help with their Rust expertise, and help us migrate some Ruby services over to Rust. This can improve the stability and performance of our product, lower costs infrastructure costs etc. What do you [client team] think?) 

Broader engineering scope (which can obviously be complementary to the above, but should be clearly stated):

"we're thinking of bringing in a very experienced team that can help with engineering processes and practices, including in devops and architecture, team org and hiring. This can improve our overall productivity, speed up delivery, tighten our feedback loop, decrease team churn. What do you think?"

Also: 

"assessment" doesn't mean people need to listen to us, right. It just means we're looking at things and giving recommendations, which can be listened to or ignored. But to be able to assess we need access, and the team buy-in, again.

Seems like the sticking point might be the "continuous assessment" wording? If that's loaded, I can remove it altogether.

twitter: oliverbarnes
tags: [startups, consulting]
bio: "Oliver Azevedo Barnes"
description:
  "Oliver Barnes on the importance of getting the team onboard when bringing in
  a consultancy"
og:
  image: /assets/images/posts/2024-05-21-team-buy-in/og-image.jpg
tagline: |
  <p>A consultancy engagement can be a very productive, gratifying collaborative process. That's _if_ the client's team is onboard with having it come in to help</p>

image: "/assets/images/posts/2024-05-21-team-buy-in/header-visual.jpeg"
imageAlt: "A rocket icon on a grey image background"
---

"Ugh, leadership is bringing in a bunch of consultants from Evil Co to audit the team and point out our inefficiencies… this can't be good, right?"

Consultants get a bad rep sometimes, and often for good reason, specially due to powerful top tier consultancies involved in everything from privatization of public companies to disaster relief. They're everywhere these days.

Software engineering consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.

Outsiders _can_ be helpful. External experience and a fresh view on things can be very useful.

> 

Reviewer 3: I'd remove this newline here. It all fits together as a single paragraph.

But, as with any collaboration, it's important to respect context and history, and to get the client's team onboard.

## Making sure we're welcome guests

When starting with a new client, we're mindful that we're coming in during a time of difficult change - be it in technical terms (changing stacks or merging systems, for example) or organizational ones, say when a team is expanding and re-organizing.

And, as often happens, both at once.

> We discuss that with the leadership team and with different stakeholders who might be involved in bringing us in to prepare the terrain for the engagement.

Reviewer 3: I don't think this sentence is needed.

Author: I see what you mean, but Marco asked to emphasize what we actually do in practice, besides understanding context.

## Happy path

When the client's team is bought-in, we get an environment of trust, a sweet spot where we complement the client's team seamlessly.

Context and knowledge are shared proactively, affording us smooth onboarding. Once we have some recommendations to give, these are discussed constructively. Decisions on what to implement are naturally done in consensus.

A virtuous cycle ensues, where implementing changes is progressively faster and more effective, and ideas flow freely back and forth. Together we shift into continous improvement mode. Everybody wins.

> So. Just how do you get buy-in?

Reviewer 3: ```suggestion
So. How do you get buy-in?
```

## Get the team involved early on

> The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea - building consensus towards the engagement.

Reviewer 3: ```suggestion
The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea - building trust in the engagement.
```

> Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [disagree and commit](https://en.wikipedia.org/wiki/Disagree_and_commit).

Reviewer 3: ```suggestion
Sometimes it's hard to get that trust, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [disagree and commit](https://en.wikipedia.org/wiki/Disagree_and_commit).
```

## Who invites us

Some very successful engagements happened when we were invited by people from the tech or product teams, who are closer to where the action will happen. This can help a lot in getting buy-in.

> When the invitation comes from the very top, it makes a huge difference when management goes the extra mile and talks to the teams that will directly be involved in and impacted by the engagement.

Reviewer 3: I get what this paragraph wants to communicate, but I think it could be reworded to be easier to read/parse 🤔

Author: how about this: 1c495228cb5e73456be4dafbd5ff73cfe1847b83

Whatever the scope, the earliest it is clearly communicated, the better.

People can become territorial over their work. Nobody likes having someone unexpected looking over their shoulders, nor receiving unsolicited advice. 

If there's a misunderstanding over the scope of our work, that's exactly how it'll feel to the client's team.

So being aligned on scope and objectives is fundamental.

## During the engagement

Client team buy-in involves accepting to at least hear recommendations from us, and for this to continue, our recommendations have to resonate and make sense, even if they the team doesn't totally agree with them.

Addressing pain points is a great way to ensure this relevance, and build trust. We're solving real problems, removing slogs. Improving productivity. Making work more pleasurable.

Often we'll find issues that weren't raised before. Or ones where a priority and solution weren't yet agreed upon. In these cases, we study and discuss until there's enough consensus.

And then we help them implement it. As part of the team.

[Reach out](/contact/) to talk about how we can help your team.

> Find [further information](/startups/) on how we can help you to burst through

Reviewer 4: ```suggestion
Find [further information](/services/team-reinforcement/) on how we can help you to burst through
```

works better here

---

> title: "Preparing the Terrain for Sucessful Engagements"

Reviewer: ```suggestion
title: "Preparing the Terrain for Successful Engagements"
```

image: "/assets/images/posts/2024-05-21-team-buy-in/header-visual.jpeg"
imageAlt: "A rocket icon on a grey image background"
---

"Ugh, leadership is bringing in a bunch of consultants from Evil Co to audit the
team and point out our inefficiencies… this can't be good, right?"

Consultants get a bad rep sometimes, and often for good reason, specially due to
powerful top tier consultancies involved in everything from privatization of
public companies to disaster relief. They're everywhere these days.

Software engineering consultancies like Mainmatter are nothing like that, but
still "consultant" is a title with plenty of loaded meanings and associations.

In the in-house team's view, consultants are often bad news: audits,
inefficiencies, and changes suggested by _outsiders_! The natural reaction is to
resist opening up, and then implementation of adjustments, even positive ones.

The team needs a very good reason to be on board.

## Making sure we're welcome guests

Outsiders can be helpful, of course. External experience and a fresh view on
things can be very useful. But, as with any collaboration, it's important to be

> aware of local context and history.

Reviewer 5: "given" maybe instead of "local"?

When starting with a new client, we're mindful that we're many times coming in
as outsiders during a time of difficult change for them - be it in technical
terms (changing stacks or merging systems, for example) or organizational ones,
say when a team is expanding and re-organizing. And, as often happens, both
things at the same time.

> We discuss that with leadership, and different stakeholders who might be

Reviewer 5: With the leadership team

involved in bringing us, and try to work with them in preparing the terrain for
the engagement.

## Happy path

> When the client team is bought-in, we get an environment of trust, a sweet spot

Reviewer 5: client's team

where we complement the client team seamlessly.

Context and knowledge are shared proactively, affording us smooth onboarding.
Once we have some recommendations to give, these are discussed constructively.
Decisions on what to implement are naturally done in consensus.

A virtuous cycle ensues, where implementing changes is progressively faster and
more effective, and ideas flow freely back and forth. Together we shift into
continous improvement mode. Everybody wins.

So. Just how do you get buy-in?

## Get the team involved early on

The earlier the intention to bring us in is communicated, the better. Before
even making a decision, hearing the team out and addressing any concerns is a

> very good idea. Building consensus regarding the engagement.

Reviewer 5: ...good idea - Building consensus...
